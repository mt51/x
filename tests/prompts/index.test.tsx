import React from 'react';

import Prompts, { type PromptsProps, PromptProps } from '../../src/prompts';
import mountTest from '../shared/mountTest';
import rtlTest from '../shared/rtlTest';
import { fireEvent, render } from '../utils';

// Mock data
const mockData: PromptProps[] = [
  {
    key: '1',
    label: 'Label 1',
    description: 'Description 1',
    icon: <span>Icon 1</span>,
    disabled: false,
  },
  {
    key: '2',
    label: 'Label 2',
    description: 'Description 2',
    icon: <span>Icon 2</span>,
    disabled: true,
  },
];

const mockProps: PromptsProps = {
  title: 'Test Title',
  items: mockData,
  onItemClick: jest.fn(),
  prefixCls: 'custom',
};

describe('prompts', () => {
  mountTest(() => <Prompts />);
  rtlTest(() => <Prompts />);
  beforeAll(() => {
    jest.useFakeTimers();
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  it('should render the title', () => {
    const { getByText } = render(<Prompts title={mockProps.title} />);
    const titleElement = getByText(/test title/i);
    expect(titleElement).toBeInTheDocument();
  });

  it('should render the correct number of items', () => {
    const { container } = render(<Prompts items={mockProps.items} />);
    expect(container.querySelectorAll('.ant-prompts-item')).toHaveLength(
      mockData.length,
    );
  });

  it('should render the labels and descriptions', () => {
    const { getByText } = render(<Prompts {...mockProps} />);
    mockData.forEach((item) => {
      const label = getByText(item.label as string);
      const description = getByText(item.description as string);
      expect(label).toBeInTheDocument();
      expect(description).toBeInTheDocument();
    });
  });

  it('should call onItemClick when a button is clicked', () => {
    const { getByText } = render(<Prompts {...mockProps} />);
    const button = getByText(/label 1/i);
    fireEvent.click(button);
    expect(mockProps.onItemClick).toHaveBeenCalledWith({ data: mockData[0] });
  });

  it('should disable buttons correctly', () => {
    const { container } = render(<Prompts items={mockProps.items} />);
    expect(container.querySelector('.ant-prompts-item-disabled')).toBeTruthy();
  });

  it('should render icons', () => {
    const { getByText } = render(<Prompts items={mockProps.items} />);
    mockData.forEach((item) => {
      if (item.icon) {
        const icon = getByText(`Icon ${item.key}`);
        expect(icon).toBeInTheDocument();
      }
    });
  });
});
