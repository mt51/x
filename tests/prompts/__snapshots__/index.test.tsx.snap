// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prompts rtl render component should be rendered correctly in RTL direction 1`] = `
<div
  class="ant-prompts ant-prompts-rtl"
>
  <div
    class="ant-prompts-list"
  />
</div>
`;
